### DISCUSSION TITLE
[v8] Sticky column pinning

### DISCUSSION THREAD

**Discussion by dcworldwide** [TanStack/table#4204] [Q&A]
Is there an example in v8, for creating pinning columns, so that when I scroll to the right, the pinned columns remain visible always (sticky)?

**Comment by LinnJS** [comment]
Tanner made a comment on this in another discussion [here](https://github.com/TanStack/table/discussions/3826) where he mentions this is not in the scope of TanStack Table as a problem to solve. However he did mention there is a sticky feature in [TanStack Virtual](https://tanstack.com/virtual/v3). 

I need this for the table I am working on as well but I think I am going to accomplish it using relative positioning and will come back to Virtual later I do want to use Virtualize List along with TanStack Query lazy load on scroll feature at some point but will wrap back around to it later.

**Reply by dcworldwide** [reply]
Thanks for all the details. That was very helpful.

 If I get it working with virtual will post here (assuming there is something worth posting)

**Reply by KevinVandy** [reply]
Its definitely possible in v8. You just have to add position sticky css the columns themselves

https://www.material-react-table.dev/?path=/story/features-column-pinning-examples--column-pinning-initial

**Reply by dcworldwide** [reply]
Is there a specific file in that repo, that shows how to do thisd, without having to use material-react-table?
